src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorConfigurationService.cs

```diff
@@ -84,25 +74,48 @@ public async override Task<RazorLSPOptions> GetLatestOptionsAsync(CancellationTo
             }
         }
 
-        private RazorLSPOptions BuildOptions(IConfiguration config)
+        private static RazorLSPOptions BuildOptions(JObject[] result)
         {
             var instance = RazorLSPOptions.Default;
 
-            Enum.TryParse(config["razor:trace"], out Trace trace);
+            var razor = result[0];
+            var html = result[1];
+            var editor = result[2];
+
+            var trace = instance.Trace;
+            if (razor.TryGetValue("trace", out var parsedTrace))
```

Comment:
For clarity could you build out two methods to represent VSCode's options and VS'? Aka

```C#
private static RazorLSPOptions BuildOptions(JObject[] result)
{
    ....
    UpdateVSCodeOptions(...);
    UpdateVSOptions(...)
    ...
}
```